src/neptune/new/internal/operation.py

```diff
@@ -197,21 +202,16 @@ def of_file(cls, value: File, attribute_path: List[str], upload_path: Path):
                 file_path=os.path.abspath(value.path),
             )
         elif value.file_type in (FileType.IN_MEMORY, FileType.STREAM):
-            tmp_file_path = cls.get_upload_path(attribute_path, value.extension, upload_path)
-            value._save(tmp_file_path)
-            operation = UploadFile(
-                path=attribute_path,
-                ext=value.extension,
-                file_path=os.path.abspath(tmp_file_path),
-                clean_after_upload=True,
-            )
+            tmp_file_name = cls.get_tmp_file_name(attribute_path, value.extension)
+            value._save(upload_path / tmp_file_name)
+            operation = UploadFile(path=attribute_path, ext=value.extension, tmp_file_name=tmp_file_name)
         else:
             raise ValueError(f"Unexpected FileType: {value.file_type}")
         return operation
 
-    def clean(self):
-        if self.clean_after_upload:
-            os.remove(self.file_path)
+    def clean(self, upload_path: Path):
```

Comment:
Why does it need to take this path as an argument if it stores path by itself? Very confusing.

Reply:
In case when we actually want to remove file we store only file name.